### ISSUE TITLE
Extremely Large Models Not Saving Properly

### ISSUE THREAD

**Issue by TylerBalsam** [tflearn/tflearn#268]
So I've been working on a very long meshed highway network with a vgg-18 perceptual loss to boot.  However, adding just a few more commands for a RGB to HSV conversion in the loss causes the whole thing to be able to train, but it totally crashes and burns during the forward pass of the trained network. I get lots of errors similar to [[Node: save_1/restore_slice_662 = RestoreSlice[dt=DT_FLOAT, preferred_shard=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](_recv_save_1/Const_0, save_1/restore_slice_662/tensor_name, save_1/restore_slice_662/shape_and_slice)]]
E tensorflow/core/client/tensor_c_api.cc:485] Tensor name "block343/b" not found in checkpoint files model.tflearn with AdaGrad, same thing for Adam.

Turning it off fixes it, turning it on breaks it. Not sure why. Here's the offending code.

        hsv_content = shift_img
        hsv_content = hsv_content + tf.ones_like(hsv_content)
        hsv_content = tf.div(hsv_content, 2.)
        hsv_content = tf.squeeze(hsv_content)
        hsv_max = tf.reduce_max(hsv_content, 2)
        hsv_min = tf.reduce_min(hsv_content, 2)
        hsv_delta = hsv_max - hsv_min
        hsv_c_sat = tf.maximum(hsv_max, 1e-8)
        hsv_sat = tf.clip_by_value(hsv_delta/hsv_c_sat, 0., 1.)
        hsv_sat = tf.expand_dims(hsv_sat, 2)
        hsv_val = tf.expand_dims(hsv_max, 2)
        hsv_content = tf.concat(2, [hsv_sat, hsv_val])
        shape = hsv_content.get_shape().as_list()
        hsv_compare = tf.concat(2, [tf.ones([shape[0], shape[1], 1]), tf.ones([shape[0], shape[1], 1])])
        variation_loss = variation_weight * tf.reduce_sum(tf.square(hsv_compare - hsv_content))

It's not the most efficient way of doing things (Hello tf.ones_like), but for testing purposes it seems to be what I need, which is creating a mean squared loss from an all 1s distribution on the SV planes of the HSV map and the input image.

Training this on one image and running it doesn't cause issues.

**Comment by aymericdamien** [comment]
What batch size are you using for training and inferencing? Maybe try to reduce it.

**Comment by TylerBalsam** [comment]
5. I reduced it from 10, and I also shaved a few layers off the main
network and still am getting it. I think something is up with the model
saver, I could run a test image forward before saving to see if it works.

**Comment by TylerBalsam** [comment]
Ok, I embedded the forward network pass on some images after the model is trained but before it is saved. That works splendidly. However, loading the same exact network from the model fails terribly. It looks more like this is a model saver issue, which if I understand correctly has had some patches to it in the past.

**Comment by aymericdamien** [comment]
Yeah, that is weird. Could you try to use TensorFlow tf.train.Saver() directly to see if the problem is related to TFLearn or TensorFlow?
